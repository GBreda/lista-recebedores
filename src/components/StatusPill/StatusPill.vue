<template>
  <div class="status-pill" :class="statusList.classColor">
    <p class="status-pill--text">{{ statusList.label }}</p>
  </div>
</template>

<script setup>
import { computed } from 'vue'

const props = defineProps({
  status: {
    type: String,
    required: true
  }
})

const statusList = computed(() => {
  if (props.status === 'rascunho') {
    return  {
      label: 'Rascunho',
      classColor: 'status-pill-draft'
    }
  }

  return {
    label: 'Validado',
    classColor: 'status-pill-validated'
  }
})
</script>

<style lang="scss" scoped>
.status-pill {
  width: 100%;
  border-radius: 2rem;
  padding: 0.7rem;

  &--text {
    text-align: center;
    color: $white;
    font-size: 1.4rem;
  }

  &-draft {
    background-color: $darken-gray;
  }

  &-validated {
    background-color: $blue;

  }
}

</style>