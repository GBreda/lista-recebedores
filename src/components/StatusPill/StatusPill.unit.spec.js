import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import StatusPill from './StatusPill.vue'

function mountStatusPill({ props } = {}) {
  return mount(StatusPill, {
    props: {
      ...props
    }
  })
}

describe('StatusPill', () => {
  it('should mount the component', () => {
    const wrapper = mountStatusPill({
      props: {
        status: 'rascunho'
      }
    })

    expect(wrapper.vm).toBeDefined()
  })

  it('should match the snapshot', () => {
    const wrapper = mountStatusPill({
      props: {
        status: 'rascunho'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders "Rascunho" status correctly', () => {
    const wrapper = mountStatusPill({
      props: {
        status: 'rascunho'
      }
    })

    const pill = wrapper.find('.status-pill')
    const text = wrapper.find('.status-pill--text')

    expect(pill.classes()).toContain('status-pill-draft')
    expect(text.text()).toBe('Rascunho')
  })

  it('renders "Validado" status correctly', () => {
    const wrapper = mountStatusPill({
      props: {
        status: 'validado'
      }
    })

    const pill = wrapper.find('.status-pill')
    const text = wrapper.find('.status-pill--text')

    expect(pill.classes()).toContain('status-pill-validated')
    expect(text.text()).toBe('Validado')
  })
})
