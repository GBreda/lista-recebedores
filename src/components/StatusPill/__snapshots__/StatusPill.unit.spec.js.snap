// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusPill > should match the snapshot 1`] = `
<div
  class="status-pill status-pill-draft"
  data-v-0df7919f=""
>
  <p
    class="status-pill--text"
    data-v-0df7919f=""
  >
    Rascunho
  </p>
</div>
`;
